CMakeLists.txt

```diff
@@ -3055,6 +3057,29 @@ if(SDL_CCACHE)
   endif()
 endif()
 
+if (SDL_CLANG_TIDY)
+  cmake_minimum_required(VERSION 3.6)
+  find_program(CLANG_TIDY_BINARY clang-tidy)
+
+  if(CLANG_TIDY_BINARY)
+    set(HAVE_CLANG_TIDY ON)
+    get_3rd_party_source(3RD_PARTY_SOURCES)
+    get_clang_tidy_ignored_files(CLANG_TIDY_IGNORED_FILES "${3RD_PARTY_SOURCES}")
+    set(CLANG_TIDY_COMMAND "${CLANG_TIDY_BINARY}" "-extra-arg=-Wno-unknown-warning-option" "--line-filter=[${CLANG_TIDY_IGNORED_FILES}]")
+    if(SDL_WERROR)
+      list(APPEND CLANG_TIDY_COMMAND "--warnings-as-errors=*")
+    endif()
+    set(CMAKE_C_CLANG_TIDY ${CLANG_TIDY_COMMAND})
+    set(CMAKE_CXX_CLANG_TIDY ${CLANG_TIDY_COMMAND})
+    set(CMAKE_OBJC_CLANG_TIDY ${CLANG_TIDY_COMMAND})
+    set_source_files_properties(${SOURCE_FILES} PROPERTIES SKIP_PRECOMPILE_HEADERS TRUE)
+    file(GLOB STDLIB_SOURCES "${SDL3_SOURCE_DIR}/src/stdlib/*.c")
+    set_source_files_properties(${STDLIB_SOURCES} PROPERTIES COMPILE_DEFINITIONS "SDL_DISABLE_ANALYZE_MACROS")
```

Comment:
Use 
```cmake
set_property(SOURCE ${STDLIB_SOURCES} APPEND PROPERTY COMPILE_DEFINITIONS "SDL_DISABLE_ANALYZE_MACROS")
```
`set_source_files_properties` overwrites properties that might have been set earlier.